### PR TITLE
GRW-1412 POC Upgrade to React 18  ⚛️

### PR DESCRIPTION
**Pull request by KarnellSchultz**
## Describe your changes

Upgrade all packages within Racoon to [React 18](https://reactjs.org/blog/2022/03/29/react-v18.html)

In general some possible concerns are around our major dependencies i.e. [Next](https://nextjs.org/docs/advanced-features/react-18/overview), Apollo Client, Emotion, Storybook

To read more about our motivations, check out the Notion document [here](https://www.notion.so/hedviginsurance/Use-React-18-for-Coverage-Store-a182a7a30cae433dbc075228bf577dda).

- Upgraded the following packages to work with React 18
- `react`
- `react-dom`
- `@react/types`
- `@react-dom/types`

## Justify why they are needed

Proactively keeping this repository up to date.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

[GRW-1412]: https://hedvig.atlassian.net/browse/GRW-1412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/ui/src/components/Menu/SubMenu.tsx [hunk 1]
```diff
@@ -63,7 +62,7 @@ type ChildrenBlurProps = {
 }
 const ChildrenBlur = ({ children, onBlur, ...props }: ChildrenBlurProps) => {
   const handleBlur = useCallback(
-    (e) => {
+    (e: React.FocusEvent<HTMLDivElement, Element>) => {
```
Comment (robinandeer) [inline packages/ui/src/components/Menu/SubMenu.tsx:65]: nit: code scout this and rename to "event" instead of "e"

**Review by robinandeer — APPROVED** [review]
Looks good and seems to build at least hehe

[excited](https://app.graphite.dev/macros/HedvigInsurance)

File: packages/ui/src/components/Menu/SubMenu.tsx (hunk 1 above)
Comment (KarnellSchultz) [reply packages/ui/src/components/Menu/SubMenu.tsx:65]: What's "code scout" mean? look on the internet for a better typing??? 🤔

File: package.json [hunk 2]
```diff
@@ -41,6 +41,6 @@
     "turbo": "1.4.2"
   },
   "resolutions": {
-    "next-i18next/@types/react": "17.0.48"
+    "next-i18next/@types/react": "18.0.17"
```
Comment (alebedev) [inline package.json:44]: You can remove this line completely, works without it

**Review by alebedev — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (KarnellSchultz) [reply package.json:44]: Right on 🚀

File: packages/ui/src/components/Menu/SubMenu.tsx (hunk 1 above)
Comment (robinandeer) [reply packages/ui/src/components/Menu/SubMenu.tsx:65]: Hmm I can't find a good definition now but we have use it as:

When implementing a new feature, try to clean up code in the vicinity.

makes sense?

Comment (KarnellSchultz) [reply packages/ui/src/components/Menu/SubMenu.tsx:65]: Yeah totally 👍 

Thanks!

Comment (simonauner) [reply packages/ui/src/components/Menu/SubMenu.tsx]: If you want to Google it I think "boy scout rule" will give you some reading material :)
https://matheus.ro/2017/12/11/clean-code-boy-scout-rule/

**Pull request merged**